### DISCUSSION TITLE
Understanding if we are at the root node in patcher

### DISCUSSION THREAD

**Discussion by Arttii** [antonmedv/expr#630] [Q&A]
Hi,

Is there an easy way to understand if we are currently visiting the root node from inside a patcher? 

I need to add a global condition to the entire statement. I guess i could patch the program itself afterwards, but was wondering if there is a more ergonomic way to do this.

Thank you

**Comment by mdmcconnell** [comment]
At some point, I tried to resolve this issue by adding a isChildNode bool to my Patcher struct, and set it true at the first Patch call.  Does that work, or can the same Patcher make multiple passes over the root?

**Comment by antonmedv** [comment]
Simply check if vm.Program.Node is equal to visited node.
